doc/how.rst

```diff
@@ -3,87 +3,160 @@
 Write a benchmark
 =================
 
-.. contents:: Contents
-   :local:
-   :depth: 2
+A benchmark consists in three elements: an objective_ function,
+a list of datasets_, and a list of solvers_.
 
 A benchmark is defined in a folder that should respect a certain
-structure. Examples of benchmarks are available in the
+structure. For example::
+
+    my_benchmark/
+    ├── README.rst
+    ├── datasets
+    │   ├── simulated.py  # some dataset
+    │   └── real.py  # some dataset
+    ├── objective.py  # contains the definition of the objective
+    └── solvers
+        ├── solver1.py  # some solver
+        └── solver2.py  # some solver
+
+Examples of actual benchmarks are available in the
 `benchOpt organisation <https://github.com/benchopt/>`_ such
 as for `Ordinary Least Square (OLS) <https://github.com/benchopt/benchmark_ols>`_,
 `Lasso <https://github.com/benchopt/benchmark_lasso>`_ or
 `L1-regularized logistic regression <https://github.com/benchopt/benchmark_logreg_l1>`_.
-The simplest way to create a benchmark is to copy an existing folder and
-modify the element to fit the new structure.
 
-A benchmark consists in three elements: an objective function,
-a list of datasets, and a list of solvers.
+.. note::
 
+    The simplest way to create a benchmark is to copy an existing folder and
+    modify the element to fit the new structure.
+
+
+.. _objective:
 
 1. Objective
 ------------
 
-The objective function is defined through a Python class that
-allows to evaluate the function that should be minimized by solvers.
-A objective should define a `set_data` method that allows
-to specify the data and a `compute` method that allows
-to evaluate the objective for a given value of the iterate.
+The **objective function** is defined through a Python class.
+This class allows to evaluate the objective function to be minimized
```

Comment:
is it necessarily minimized (eg left out loss in stochastic bench) ? 
"This class allows to monitor the quantities of interest along iterations of the solvers" ?